src/client/kv/dc_kv.c

```diff
@@ -403,6 +404,7 @@ dc_kv_put(tse_task_t *task)
 	rc = tse_task_register_comp_cb(task, free_io_params_cb, &params, sizeof(params));
 	if (rc != 0)
 		D_GOTO(err_utask, rc);
+	free_params = false;
```

Comment:
The other option would be to set params to NULL here which might be safer and clearer but would mean calling D_FREE(params) with a NULL value for params later which would look odd.

Reply:
it's more clear that way.